src/FuncSharp.Examples/IOption.cs

```diff
@@ -0,0 +1,98 @@
+            divisionResult.Match(
+                result => Console.Write($"Division successful, result is: {result}."),
+                _ => Console.Write("Division failed, must have divided by zero.")
+            );
+            divisionResult.Match(result => Console.Write($"Division successful, result is: {result}."));
+            divisionResult.Match(ifSecond: _ => Console.Write("Division failed, must have divided by zero."));
+        }
+
+        private void GettingOptionValue(decimal number, decimal divisor)
+        {
+            var divisionResult = Divide(number, divisor);
+
+            // `Get` will throw an exception in case of empty option. Therefore you should not really use it at all.
+            // You should rather use Match to branch your code into individual cases where each case is guaranteed to work.
+            decimal valueOrExceptionThrown = divisionResult.Get();
+
+            decimal valueOrFallback1 = divisionResult.GetOrZero();
+            decimal valueOrFallback2 = divisionResult.GetOrElse(114m);
+            decimal valueOrFallback3 = divisionResult.GetOrElse(_ => 114m); // Lazy. Will only run the lambda if it needs to.
+
+            // These two are identical. Just Map creates one extra instance of Option for no reason.
+            decimal roundedDivisionResult1 = divisionResult.Map(r => Math.Round(r)).GetOrZero();
+            decimal roundedDivisionResult2 = divisionResult.Match(
+                r => Math.Round(r),
+                _ => 0
+            );
+        }
+
+        private void HandlingCollectionsOfOptions()
+        {
+            var numbers = new List<int> { 0, 1, 2, 3 };
+            var divisors = new List<int> { 0, 1, 2, 3 };
+            IEnumerable<IOption<decimal>> divisionResults = numbers.SelectMany(n => divisors.Select(d => Divide(n, d))).ToList();
+            // These two lines produce equal results. But flatten is more readable and generally using `Get` is an anti-pattern as it is not safe.
+            IEnumerable<decimal> successfulResults1 = divisionResults.Flatten();
+            IEnumerable<decimal> successfulResults2 = divisionResults.Where(r => r.NonEmpty).Select(r => r.Get()); // get throws exception when called on empty option
+
+            int errorResultCount = divisionResults.Count(r => r.IsEmpty);
+        }
+
+        private void OptionCreatingExamples()
+        {
+            IOption<bool> emptyOption1 = Option.Empty<bool>();
+            bool? emptyNullableBool = null;
+            IOption<bool> emptyOption2 = emptyNullableBool.ToOption();
+            IOption<bool> emptyOption3 = Option.Create(emptyNullableBool);
+
+            IOption<bool> valuedOption1 = Option.Valued<bool>(false);
+            bool? nullableBool = false;
+            IOption<bool> valuedOption2 = nullableBool.ToOption();
+            IOption<bool> valuedOption3 = Option.Create(nullableBool);
+
+            // Option.Valued can construct options with null value inside.
```

Comment:
I'd say here it's antipattern and it's better to use option of option for clarity as there is no ambiguity.